docs: add documentation for `nuxi module` (#24790)
nuxt/nuxt: docs/3.api/4.commands/module.md

---
title: "nuxi module"

> description: "Module utilities for your Nuxt application"

Reviewer: ```suggestion
description: "Search and add modules to your Nuxt application with the command line."
```

links:
  - label: Source
    icon: i-simple-icons-github

>     to: https://github.com/nuxt/cli/blob/main/src/commands/module/add.ts

Author: Should this be changed to /module? (As in, remove the /add.ts part because it contains 2 commands)

Reviewer: Yes

>     to: https://github.com/nuxt/cli/blob/main/src/commands/module/add.ts

Reviewer: ```suggestion
    to: https://github.com/nuxt/cli/blob/main/src/commands/module/
```

Reviewer 2: We might even remove the redundant trailing slash :)

    size: xs
---

Nuxi provides a few utilities to work with [Nuxt modules](/modules) seamlessly.

## nuxi module add

```bash [Terminal]
npx nuxi module add <NAME>
```

Option        | Default          | Description
-------------------------|-----------------|------------------
`NAME` | - | The name of the module to install.

> The command lets you install [Nuxt modules](/modules) to your application with no manual work.

Reviewer 2: ```suggestion
The command lets you install [Nuxt modules](/modules) in your application with no manual work.
```

When running the command, it will:

> - install the module as dependency using your package manager

Author: ```suggestion 
- install the module as a dependency using your package manager
```

> - add it to your [package.json](/docs/guide/directory-structure/package)

Author: ```suggestion
- add it to your [package.json](/docs/guide/directory-structure/package) file.
```

Reviewer 3: Do we need to specify this (or maybe before the installation) ? It should also update the lock file (not sure if it is relevant to mention it tho)

Reviewer: I think the lock file might be too much to add but I still think it's good to mention the package.json

Author: @huang-julien technically I suppose both are valid options, we can either specify everything it does or mention the most important things

Author: In chronological order it first installs before overwriting package.json, so I don't think there's a reason to mention it *before* the installation. (Although it does both do it doesn't really matter that much, as you can add packages that aren't installed to package.json, and I guess you'll just receive an error)

> - add it to your [package.json](/docs/guide/directory-structure/package) file.
> - update your [`nuxt.config`](/docs/guide/directory-structure/nuxt-config) file.

Reviewer: ```suggestion
- add it to your [package.json](/docs/guide/directory-structure/package) file
- update your [`nuxt.config`](/docs/guide/directory-structure/nuxt-config) file
```

> - update your [`nuxt.config`](/docs/guide/directory-structure/nuxt-config) file with each module you add.

Author: ```suggestion
- update your [nuxt.config](/docs/guide/directory-structure/nuxt-config) file.

**Example:**

Installing the [`Pinia`](/modules/pinia) module
```bash [Terminal]
npx nuxi module add pinia 
```

## nuxi module search

```bash [Terminal]
npx nuxi module search <QUERY>
```

Option        | Default          | Description
-------------------------|-----------------|------------------
`QUERY` | - | The name of the module to search for.

> The command searchs for Nuxt modules matching your query, compatible with your current Nuxt version installed.

Author: ```suggestion 
The command searches for Nuxt modules matching your query, compatible with your Nuxt version.
```

---
title: "nuxi module"
description: "Module utilities for your Nuxt application"
links:
  - label: Source
    icon: i-simple-icons-github
    to: https://github.com/nuxt/cli/blob/main/src/commands/module/add.ts
    size: xs
---

Nuxi provides a few utilities to work with [Nuxt modules](/modules) seamlessly.

## nuxi module add

```bash [Terminal]
npx nuxi module add <NAME>

> ```

Reviewer: ```suggestion
```

```

The `module add` command allows you to install [Nuxt modules](/modules) directly to your application with minimal manual work. It automatically updates your [`nuxt.config`](/docs/guide/directory-structure/nuxt-config) file with each module you add.

Option        | Default          | Description
-------------------------|-----------------|------------------
`NAME` | - | The name of the module to install.

> ### Example usage

Reviewer: ```suggestion
### Example
```

Installing the [`Pinia`](/modules/pinia) module
```bash [Terminal]
npx nuxi module add pinia 
```

> Your `nuxt.config` should now look like this:

Reviewer 3: ```suggestion
Your `nuxt.config` will be modified to:
```

```ts [nuxt.config.ts]
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: ['@pinia/nuxt']
})
```

## nuxi module search

```bash [Terminal]
npx nuxi module search <QUERY>
```
The `module search` command allows you to search for a Nuxt module directly from your application.

Option        | Default          | Description
-------------------------|-----------------|------------------
`QUERY` | - | The name of the module to search for.

> ### Example usage

Reviewer: ```suggestion
### Example
```
